File: backend/src/main/java/com/ums/ums_backend/entity/Professor.java
package com.ums.ums_backend.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

/**
 * Represents a professor teaching courses with personal information and course instructor assignments.
 */
@Entity
@Getter
@Setter
public class Professor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name="user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 30)
    private String firstName;

    @Column(nullable = false, length = 30)
    private String lastName;

    @Enumerated(EnumType.STRING)
    private Gender gender;

    @Column(nullable = false)
    private LocalDate dateOfBirth;

    @Column(nullable = false, unique = true)
    private String email;

    @OneToMany(mappedBy = "professor")
    private List<CourseInstructor> courseInstructors;


}

File: frontend/src/app/features/professor/models/professor.ts
import {Gender} from '../../../core/enums/Gender';
import {CourseInstructor} from '../../courseInstructor/models/course-instructor';

export interface Professor {

  id: number,
  userId: number,
  firstName: string,
  lastName: string,
  gender: Gender,
  dateOfBirth: string,
  email: string,
  courses?: CourseInstructor[]

}

export interface ProfessorCreateRequest {

  firstName: string,
  lastName: string,
  gender: Gender,
  dateOfBirth: string,
  email: string,

}
